Assets/Scripts/LevelSelect/Player/Vehicle.cs

```diff
@@ -0,0 +1,150 @@
+        [SerializeField] private Transform body;
+
+        private float speed, speedTarget;
+        private float rotate, rotateTarget;
+
+        private bool canSteer = false;
+        private bool nearGround, onGround;
+
+        private Vector3 containerBase;
+
+        void Awake()
+        {
+            container = vehicleModel.GetChild(0);
+            containerBase = container.localPosition;
+        }
+
+        void Update()
+        {
+            Accelarate();
+            WheelAndBodyTilt();
+            VehicleTilt();
+            Steering();
+
+            // Stops vehicle from floating around when standing still
+
+            if (speed == 0 && sphere.velocity.magnitude < 4f)
+            {
+                sphere.velocity = Vector3.Lerp(sphere.velocity, Vector3.zero, Time.deltaTime * 2.0f);
+                canSteer = false;
+            }
+            else
+            {
+                canSteer = true;
+            }
+        }
+
+        void FixedUpdate()
+        {
+            MovementHandler();
+        }
+
+        private void VehicleTilt()
+        {
+            float tilt = 0.0f;
+
+            container.localPosition = containerBase + new Vector3(0, Mathf.Abs(tilt) / 2000, 0);
+            container.localRotation = Quaternion.Slerp(container.localRotation, Quaternion.Euler(0, rotateTarget / 8, tilt), Time.deltaTime * 10.0f);
+        }
+
+        private void WheelAndBodyTilt()
+        {
+            if (wheelFrontLeft != null)
+            {
+                wheelFrontLeft.localRotation = Quaternion.Euler(0, rotateTarget / 2, 0);
+            }
+
+            if (wheelFrontRight != null)
+            {
+                wheelFrontRight.localRotation = Quaternion.Euler(0, rotateTarget / 2, 0);
+            }
+
+            body.localRotation = Quaternion.Slerp(body.localRotation, Quaternion.Euler(new Vector3(speedTarget / 4, 0, rotateTarget / 6)), Time.deltaTime * 4.0f);
+        }
+
+        private void Accelarate()
+        {
+            speedTarget = Mathf.SmoothStep(speedTarget, speed, Time.deltaTime * 12f);
+
+            speed = 0f;
+
+            if (Input.GetKey(KeyCode.W))
+            {
+                speed = acceleration;
+            }
+
+            if (Input.GetKey(KeyCode.S))
+            {
+                speed = -acceleration;
+            }
+        }
+
+        private void Steering()
+        {
+            rotateTarget = Mathf.Lerp(rotateTarget, rotate, Time.deltaTime * 4f); rotate = 0f;
+
+            if (canSteer && nearGround)
+            {
+
+                if (Input.GetKey(KeyCode.A))
+                {
+                    rotate = -steering;
+                }
+
+                if (Input.GetKey(KeyCode.D))
+                {
+                    rotate = steering;
+                }
+
+            }
+
+            transform.rotation = Quaternion.Slerp(transform.rotation, Quaternion.Euler(new Vector3(0, transform.eulerAngles.y + rotateTarget, 0)), Time.deltaTime * 2.0f);
```

Comment:
Werkt het sturen ook hetzelfde als de FPS te laag is? (of te hoog?)

Reply:
yes